From 7e7836d6a51eac0af6cfe0023e32ffc936213703 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:09:19 -0400
Subject: [powerpc] powernv: Remove get/set_rtc_time when they are not present

Message-id: <1410545655-205645-331-git-send-email-dzickus@redhat.com>
Patchwork-id: 94283
O-Subject: [RHEL7 PATCH 330/626] powernv: Remove get/set_rtc_time when they are not present
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 2c49195b6aedd21ff1cd1e095fab9866fba3411b
Author: Michael Neuling <mikey@neuling.org>
Date:   Fri Nov 29 14:22:48 2013 +1100

    powernv: Remove get/set_rtc_time when they are not present

    Currently we continue to poll get/set_rtc_time even when we know they
    are not working.

    This changes it so that if it fails at boot time we remove the ppc_md
    get/set_rtc_time hooks so that we don't end up polling known broken
    calls.

    Signed-off-by: Michael Neuling <mikey@neuling.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/powernv/opal-rtc.c b/arch/powerpc/platforms/powernv/opal-rtc.c
index 7d07c7e..b1885db 100644
--- a/arch/powerpc/platforms/powernv/opal-rtc.c
+++ b/arch/powerpc/platforms/powernv/opal-rtc.c
@@ -18,6 +18,7 @@
 
 #include <asm/opal.h>
 #include <asm/firmware.h>
+#include <asm/machdep.h>
 
 static void opal_to_tm(u32 y_m_d, u64 h_m_s_ms, struct rtc_time *tm)
 {
@@ -48,8 +49,11 @@ unsigned long __init opal_get_boot_time(void)
   else
    mdelay(10);
  }
- if (rc != OPAL_SUCCESS)
+ if (rc != OPAL_SUCCESS) {
+  ppc_md.get_rtc_time = NULL;
+  ppc_md.set_rtc_time = NULL;
   return 0;
+ }
  y_m_d = be32_to_cpu(__y_m_d);
  h_m_s_ms = be64_to_cpu(__h_m_s_ms);
  opal_to_tm(y_m_d, h_m_s_ms, &tm);
-- 
1.7.1